Data-Structures/Vectors/Vector2.test.js

```diff
@@ -0,0 +1,58 @@
+import { Vector2 } from './Vector2.js'
+
+describe('Tests for Vector2 data scructure implementation', () => {
+  test('tests equalsExactly-method', () => {
```

Comment:
Nice job! Now this is me being nitpicky: I always try to group all things related to one test target (e.g. a specific method) in their own `describe` block. As you can see from my version, it's totally fine to nest `describes`.

It's not a _must_ but I think it makes Jest's console output a bit nicer:

```
 PASS  Data-Structures/Vectors/Vector2.test.js
  Vector2
    #equalsExactly
      ✓ should compare equality correctly (2 ms)
    #equalsApproximately
      ✓ should compare equality (approximately) correctly
    #add
      ✓ should add two vectors correctly (1 ms)
    #subtract
      ✓ should subtract two vectors correctly
    #multiply
      ✓ should multiply two vectors correctly
    #length
      ✓ should calculate it's length correctly
    #normalize
      ✓ should normalize vectors correctly
    #distance
      ✓ should calculate the distance between two vectors correctly (1 ms)
    #dotProduct
      ✓ should calculate the dot product correctly
    #rotate
      ✓ should rotate a vector correctly (1 ms)
    #angleBetween
      ✓ should calculate the angle between two vectors correctly
```

Reply:
I also think it pleases the output. And as I realized if I would change my code to match this approach I would just be typing what you already typed, so here I just straight up copied your code. Hope that's ok!